### PR TITLE
Updates suggested client version from 510 to 511

### PR DESCRIPTION
**Pull request by Kyep**
🆑Kyep
tweak: Changed "suggested client version" from 510 to 511.
/🆑

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Paradise doesnt work well with BYOND 511

**Comment by Kyep** [comment]
@Allfd ?

**Comment by Lady-Luck** [comment]
The server message already reflects the 511 change so there shouldn't be any need for deliberation:

'We've updated to 511.1379, for the best experience you should to.'

**Comment by tigercat2000** [comment]
@Kyep Can you include an update to .travis.yml in this PR, please? 

Change these two lines
https://github.com/ParadiseSS13/Paradise/blob/master/.travis.yml#L10-L11
```
    - BYOND_MAJOR="510"
    - BYOND_MINOR="1346"
```
to
```
    - BYOND_MAJOR="511"
    - BYOND_MINOR="1379"
```

**Comment by Allfd** [comment]
@Kyep @AffectedArc07 

I recommend clearing the cache.  Very odd occurrence.

**Comment by Anticept** [comment]
Be aware that 511 should have fixed the goonlighting problem with the memleak and CPU usage too.

> A memory leak was found for color matrices in certain cases. (MrStonedOne)

http://www.byond.com/forum/?post=2220974

**Comment by Kyep** [comment]
@tigercat2000 Done.

**Comment by taukausanake** [comment]
I haven't updated yet. Should I now?

**Comment by Lady-Luck** [comment]
>Be aware that 511 should have fixed the goonlighting problem with the memleak and CPU usage too.

We probably shouldn't get rid of the draemdaemon relaunch script though, it's sped up server restarts significantly.

**Comment by Krausus** [comment]
Uhh, as far as I'm aware, there's no pressing reason to tell players to update to 511. It's still in beta, and we should be *avoiding* beta releases whenever possible, especially with some of those beta releases having caused serious problems client-side. This...

> Be aware that 511 should have fixed the goonlighting problem with the memleak

... is why we've already upgraded the **server** to 511. There's also some other server-side fixes that are useful.

> The server message already reflects the 511 change so there shouldn't be any need for deliberation

Yeah... @Allfd did that, probably not realizing that it's still in beta and has had the aforementioned client-side problems. We don't even use any 511 features that would require an updated client yet. That should probably not be in the MOTD.

**Comment by Lady-Luck** [comment]
>Uhh, as far as I'm aware, there's no pressing reason to tell players to update to 511. It's still in beta, and we should be avoiding beta releases whenever possible, especially with some of those beta releases having caused serious problems client-side. This...

511 is currently more stable than stable for both the client and the server.

**Comment by Krausus** [comment]
> 511 is currently more stable than stable for both the client and the server.

I'm sure people said the same thing about the 511 releases that preceded the ones which broke browser controls to various extents. "Beta" means "things can (and will) break", and we've *already* had to deal with people upgrading to beta releases that break things.

> Stacking progress bars do not work properly in 510 and are already implemented.

But... that doesn't make any sense. They shouldn't need any 511-specific feature. We explicitly don't use any 511-specific features. How do they not work in 510?

**Comment by Lady-Luck** [comment]
>But... that doesn't make any sense. They shouldn't need any 511-specific feature. We explicitly don't use any 511-specific features. How do they not work in 510?

A bug.

**Comment by Anticept** [comment]
As I said above, there is a *critical fix* in 511.

**Comment by Krausus** [comment]
> A bug that is fixed in 511.

Of course someone would have developed a feature that doesn't work properly in the current stable version. I should have guessed.

Well... suggesting that people upgrade to 511 is probably fine. There's no way clients could end up broken *again,* right?

Would probably be a good idea to update the warning message, though, so it includes the version the server recommends and mentions that it may be a beta version.

**Comment by Spacemanspark** [comment]
>Stacking progress bars do not work properly in 510 and are already implemented.

...Yes they do, I'm still using 510 and it works just fine.

**Comment by Anticept** [comment]
@Krausus, it's to do with the color matrix cache in byond. There was a memleak which was causing server crashes.

**Comment by taukausanake** [comment]
So it's only the server that needed the update. The clients don't really need it, do they?

Only thing I noticed on 511 is that the colors look a _little_ different in it. More saturated maybe? I'm not good with colors but there is a difference it appearance

**Comment by Anticept** [comment]
It also affects the client, but not as badly. In addition, 511 checks directx.

Finally, mismatching client and server can result in weird bugs.

**Comment by Krausus** [comment]
> @Krausus, it's to do with the color matrix cache in byond.

@Anticept, I'm well aware of that fix. We're already running 511 on the server. This PR is completely unrelated to that - it causes people using the current stable client (or older) to get a warning that they should update.

> > Stacking progress bars do not work properly in 510 and are already implemented.
> 
> ...Yes they do, I'm still using 510 and it works just fine.

Huh. Yeah, they seem to work fine on 510. Welp.

> It also affects the client, but not as badly.

There's a different client-side memory leak that was fixed in 511, but I'm not sure it was severe enough for anyone to really bump into it. The server-side leak doesn't affect clients.

> In addition, 511 checks directx.

I don't see this mentioned in the 511 release notes. Are you sure you aren't thinking about the DirectInput support that was added for gamepads?

> Finally, mismatching client and server can result in weird bugs.

BYOND has always focused pretty heavily on maintaining backwards compatibility, for some reason. Old clients generally aren't an issue until you start using features they don't support, which we've thus far avoided for 511.

**Comment by Anticept** [comment]
> There's a different client-side memory leak that was fixed in 511, but I'm not sure it was severe enough for anyone to really bump into it. The server-side leak doesn't affect clients.

Different but closely related. Lummox was talking about appearances on the client side as part of the bug report, but he was unsure where exactly the problem is. It's not so much about the critical fix, but keeping client and server matched.

> BYOND has always focused pretty heavily on maintaining backwards compatibility, for some reason. Old clients generally aren't an issue until you start using features they don't support, which we've thus far avoided for 511.

It's still byond, and they've had a running track record of weird quirks when client has been ahead of server. Whether that's just because of beta or because of something under the hood, it's still really annoying trying to chase down problems and finding out it's a client issue.

File: .travis.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ git:
 
 env:
   global:
-    - BYOND_MAJOR="510"
-    - BYOND_MINOR="1346"
+    - BYOND_MAJOR="511"
+    - BYOND_MINOR="1379"
```
Comment (Krausus) [inline .travis.yml]: Will want this to be 1380, though, since that's the minor version for the new stable release.

**Comment by Fox-McCloud** [comment]
@Krausus will you be pleased with this once it's updated to 511.1380?

**Comment by Krausus** [comment]
> will you be pleased with this once it's updated to 511.1380?

Yeah; 1380 is the new stable, and includes an extra warning (for assignment operator misuse) that we'll want to have show up in CI.

**Comment by Anticept** [comment]
Version is now 511.1381

**Comment by Allfd** [comment]
This should probably be a config option

**Comment by Fox-McCloud** [comment]
Please update.

**Comment by Kyep** [comment]
Updated to 511.1380, the new stable. @Fox-McCloud @Krausus

**Comment by taukausanake** [comment]
> Version is now 511.1381

Huh... Unless server binaries are behind a tad?

**Comment by Krausus** [comment]
Eh, 511.1381 *is* the latest version now, but for the purpose of CI, it doesn't change anything that matters. This is fine as-is.

**Pull request merged**